src/uu/mknod/src/mknod.rs

```diff
@@ -5,22 +5,42 @@
 // For the full copyright and license information, please view the LICENSE
 // file that was distributed with this source code.
 
-// spell-checker:ignore (ToDO) parsemode makedev sysmacros makenod newmode perror IFBLK IFCHR IFIFO
-
-mod parsemode;
+// spell-checker:ignore (ToDO) parsemode makedev sysmacros makenod perror IFBLK IFCHR IFIFO
 
 #[macro_use]
 extern crate uucore;
 
+use std::ffi::CString;
+
+use clap::{App, Arg, ArgMatches};
 use libc::{dev_t, mode_t};
 use libc::{S_IFBLK, S_IFCHR, S_IFIFO, S_IRGRP, S_IROTH, S_IRUSR, S_IWGRP, S_IWOTH, S_IWUSR};
 
-use getopts::Options;
-
-use std::ffi::CString;
+mod parsemode;
 
 static NAME: &str = "mknod";
 static VERSION: &str = env!("CARGO_PKG_VERSION");
+static USAGE: &str = "
+Usage: mknod [OPTION]... NAME TYPE [MAJOR MINOR]
+
+Mandatory arguments to long options are mandatory for short options too.
+-m, --mode=MODE    set file permission bits to MODE, not a=rw - umask
+--help     display this help and exit
+--version  output version information and exit
+
+Both MAJOR and MINOR must be specified when TYPE is b, c, or u, and they
+must be omitted when TYPE is p.  If MAJOR or MINOR begins with 0x or 0X,
+it is interpreted as hexadecimal; otherwise, if it begins with 0, as octal;
+otherwise, as decimal.  TYPE may be:
+
+b      create a block (buffered) special file
+c, u   create a character (unbuffered) special file
+p      create a FIFO
+
+NOTE: your shell may have its own version of mknod, which usually supersedes
+the version described here.  Please refer to your shell's documentation
+for details about the options it supports.
+";
```

Comment:
I suspect this is 1:1 copied from GNU mknod. Are we allowed to do this? Otherwise we could also rely on `clap` to create this automatically..

Reply:
This shouldn't be a problem since GNU coreutils licensed with `GPL 3+`.

Reply:
I don't know a lot about licenses so I'm curious. My understanding was that if you copy stuff from GPL licensed code, the derivative must be published under GPL as well.

If you have some time, it would be nice if you could explain why it is not a problem as you said above / why my understanding is false.

Reply:
See: https://github.com/uutils/coreutils/issues/834, a lot of discussions happened on there

Reply:
TBH i was actually thinking like you, although i'm  still not 100% confident about that but it seems like it's not a problem.

Reply:
Thanks a lot for the pointer, I'll read up on it :)